### ISSUE TITLE
Intermitted expressions failures during testing

### ISSUE THREAD

**Issue by bodiam** [datafaker-net/datafaker#1190]
**Describe the bug**

It seems there's another or new bug in our tests:

```
 10:28:42.078 [ERROR] Failures: 
10:28:42.078 [ERROR] net.datafaker.providers.base.RestaurantTest.name
10:28:42.078 [ERROR]   Run 18: RestaurantTest.name:33 
Expecting actual:
  "## Bakery"
not to contain:
  ["#", "?"]
but found:
  ["#"]
```

**To Reproduce**

I cannot.

**Expected behavior**

The test should pass

**Versions:**
 - Faker Version 2.2.x-snapshot

**Comment by snuyanzin** [comment]
That looks strange, it looks like sometimes it doesn't take into account possible expressions inside values.
I'm not sure whether it is COW, currently it doesn't look like that

**Comment by bodiam** [comment]
I've ran the test locally quite a few times, and I'm not able to reproduce it, but I will have some time over the weekend to look into this.

**Comment by snuyanzin** [comment]
If I'm not mistaken @RVRhub is also looking at it, please correct me if I'm wrong

**Comment by RVRhub** [comment]
Yes, I started looking into this problem.
I have determined that this is because sometimes resolveExpression uses a Supplier that retrieves data from a file.

**Comment by snuyanzin** [comment]
@bodiam it might happen that @RVRhub has fixed it within #1198  (not 100%)
however now it looks much more stable, so if you know how to reproduce it would worth to check it against the latest main

**Comment by bodiam** [comment]
@snuyanzin no idea how to reproduce it, I just know it happens sometimes. Let's monitor it for a week or so, and then close this issue if it's fixed.
